tensorflow_probability/python/experimental/bijectors/highway_flow_test.py

```diff
@@ -0,0 +1,127 @@
+            self.assertStartsWith(bijector.name, 'highway_flow')
+            self.assertAllClose(x, bijector.inverse(tf.identity(bijector.forward(x))))
+            self.assertAllClose(bijector.forward_log_det_jacobian(x, event_ndims=dim+1),
+                                -bijector.inverse_log_det_jacobian(tf.identity(bijector.forward(x)), event_ndims=dim+1))
+
+    def testJacobianWithActivation(self):
+        #activations = ['sigmoid', 'softplus', 'tanh', 'none']
+        activations = ['softplus']
+        batch_size = 3
+        width = 4
+        dtype = tf.float32
+        residual_fraction = tf.Variable(0.5)
+        for activation in activations:
+
+            if activation == 'sigmoid':
+                activation_fn = tf.nn.sigmoid
+            elif activation == 'softplus':
+                activation_fn = tf.nn.softplus
+            elif activation == 'tanh':
+                activation_fn = tf.nn.tanh
+            elif activation == 'none':
+                activation_fn = None
+
+            bijector = tfp.experimental.bijectors.HighwayFlow(
+                width=width,
+                residual_fraction=residual_fraction,
+                activation_fn=activation_fn,
+                bias=tf.Variable(0., dtype=dtype),
+                upper_diagonal_weights_matrix=tf.Variable(tf.eye(width)),
+                lower_diagonal_weights_matrix=tf.Variable(tf.eye(width))
+            )
+
+            self.evaluate([v.initializer for v in bijector.trainable_variables])
+            x = tf.ones((batch_size, width)) # * samplers.uniform((batch_size, width), -10., 10., seed=seed)
+            if activation == 'none':
+                y = x
+            else:
+                y = residual_fraction * x + (1 - residual_fraction) * activation_fn(x)
+            expected_forward_log_det_jacobian = _activation_log_det_jacobian(x, residual_fraction, activation)
+            expected_inverse_log_det_jacobian = -expected_forward_log_det_jacobian
+            self.assertAllClose(y, bijector.forward(x))
+            self.assertAllClose(x, bijector.inverse(y))
+            self.assertAllClose(
+                expected_inverse_log_det_jacobian,
+                bijector.inverse_log_det_jacobian(y, event_ndims=1),
+            )
+            self.assertAllClose(
+                expected_forward_log_det_jacobian,
+                bijector.forward_log_det_jacobian(x, event_ndims=1),
+            )
+
+
+    def testResidualFractionGradientsWithCenteredDifference(self):
+        width = 4
+        batch_size = 3
+        dtype = tf.float32
+        residual_fraction = tf.Variable(0.5)
+        bijector = tfp.experimental.bijectors.HighwayFlow(
+            width=width,
+            residual_fraction=residual_fraction,
+            activation_fn=tf.nn.softplus,
+            bias=tf.Variable(0., dtype=dtype),
+            upper_diagonal_weights_matrix=tf.Variable(tf.eye(width)),
+            lower_diagonal_weights_matrix=tf.Variable(tf.eye(width))
```

Comment:
I don't think any of these parameters (including residual_fraction) need to be Variables, since you're not actually updating them. You can just write `residual_fraction = tf.constant(0.5)`,
`upper_diagonal_weights_matrix=tf.eye(width)`, etc. Variables are special in that they're automatically watched by gradient tapes, but you don't need that since you're already asking the tape to watch `residual_fraction` specifically.